frontend/src/types/oauth.ts

```diff
@@ -25,7 +39,14 @@ export function redirectUrl(): string {
   return `${window.location.origin}/oauth/callback`;
 }
 
-type OAuthType = "login" | "register";
+// signin: users try to login via oauth
+// register-vcs: users try to bind a vcs to her workspace
+// link-vcs-repository: users try to bind a vcs repo to her project
+export type OAuthType =
+  | "bb.oauth.signin"
+  | "bb.oauth.register-vcs"
+  | "bb.oauth.link-vcs-repository"
+  | "bb.oauth.unknown";
```

Comment:
is bb.oauth.unknown still needed?

Reply:
`bb.oauth.unknown` is just a reminder. I used `bb.oauth.unknown` to register a handler for unknown types.

Reply:
I don't think we need to do that. Instead it's better to post an error notification

Reply:
Or just throw an exception? The main point is there is no use for sending a bb.oauth.unknown since we didn't subscribe it in any place

Reply:
PTAL